src/value_generators/include/value_generators/PipelineGenerator.hpp

```diff
@@ -29,16 +29,30 @@ namespace genny {
 struct PipelineGenerator {
     PipelineGenerator() = default;
 
-    PipelineGenerator(const Node& node, PhaseContext& context, ActorId id) {
-        if (!node.isSequence()) {
-            BOOST_THROW_EXCEPTION(InvalidConfigurationException("'Pipeline' must be an array"));
+    PipelineGenerator(const Node& node, ActorContext& context) {
+        assertIsArray(node);
+        for (auto&& [_, stageNode] : node) {
+            // The '0' here is a lie, but we don't necessarily have an actor ID yet in this
```

Comment:
Oops the '0' is stale (we start at 1), but this is the part of the patch I'm not sure about. Should I make ActorId optional in the DocumentGenerator class? And then assert that it's present in the `{^ActorId` generator?

Reply:
@rtimmons looking for your input here specifically.

Reply:
This is a good question. But I'm confused about why we don't necessarily have the ActorId here when we do usually have it when constructing other generator types. Is there a usage pattern that we don't capture well?

Reply:
Well, conceptually we don't have an ActorId because we are trying to generate a single pipeline to be used to generate a single sample to be shared across all threads (which I am pretty sure all have different ActorIds).

We maybe could technically get an ActorId in here if I reworked a bunch of code to avoid even creating the PipelineGenerator until we are lazily loading the sample, but it would be a random ActorId - whichever one won the race.